workers/unity/Assets/Playground/Scripts/Cubes/CubeMovementSystem.cs

```diff
@@ -19,20 +20,25 @@ public struct Data
 
         [Inject] private Data data;
 
-        private static Vector3 speed = new Vector3(2, 0, 0);
-
-        private Vector3 origin;
-
-        protected override void OnCreateManager(int capacity)
+        public struct WorkerData
         {
-            base.OnCreateManager(capacity);
-
-            var worker = WorkerRegistry.GetWorkerForWorld(World);
-            origin = worker.Origin;
+            public readonly int Length;
+            [ReadOnly] public SharedComponentDataArray<WorkerConfig> WorkerConfigs;
         }
 
+        [Inject] private WorkerData workerData;
+
+        private static Vector3 speed = new Vector3(2, 0, 0);
+
         protected override void OnUpdate()
         {
+            if (workerData.Length == 0)
+            {
+                new LoggingDispatcher().HandleLog(LogType.Error, new LogEvent("This system should not have been run without a worker entity"));
+            }
+
+            var origin = workerData.WorkerConfigs[0].Worker.Origin;
+
```

Comment:
This boilerplate makes me sad

Reply:
me too :(
maybe we need to create a `SpatialSystem` after all that takes care of that boilerplate code.